posthog/session_recordings/sql/session_replay_event_migrations_sql.py

```diff
@@ -222,6 +222,25 @@ def DROP_KAFKA_SESSION_REPLAY_EVENTS_TABLE_SQL(on_cluster=True):
     )
 )
 
+# migration to add a TTL policy to the session replay tables
+ALTER_SESSION_REPLAY_ADD_TTL = """
+    ALTER TABLE {table_name}
+        MODIFY TTL dateTrunc('day', min_first_timestamp) + toIntervalDay(coalesce(retention_period_days, 365))
```

Comment:
so does the dateTrunc mean
if i have a recording at 00:01 and one at 23:59
then one gets effectively 1 day more retention that the other?

i wonder how users think about it 🤔 
do we _need_ the trunc?

Reply:
It's a matter of how we explain the retention. We can say; data is deleted at midnight UTC.

I don't think it's an issue, and if we put a condition in where (we should) then this won't be troublesome.

Any deletes in CH are lazy anyway. This is just optimization, all rows from given day deleted at once.

Reply:
ok... so CH would delete 00:01 and 23:59 at the same point in the same day whenever it gets to it

so 00:01 might be deleted later (which is the same as S3 behavior which makes no promises)
but 23:59 might be deleted earlier (which is _I think_ different to S3 behavior)

it'd be cool if we could make it behave how a human expects since then we don't have to explain it
and... we shouldn't delete _early_ if we can avoid it

so i'd be tempted to set the TTL in CH to 1 day more than the retention
and then we can soft limit too in the query or the UI
that way we delete a few hours late rather than early which is _probably_ better

and needs explaining way less often imho